src/Misc/Primes.jl

```diff
@@ -117,7 +117,7 @@ struct PrimesSet{T}
       p = next_prime(p, false)
     end
     mod = abs(mod)
-    val = abs(val)
+    val = abs(val % mod)
```

Comment:
I don't think this is quite right.
```
julia> -1 % 3
-1
```
So this code turns -1 mod 3 into 1 mod 3.

Reply:
mod(x, y) will always be positive if y is positive. So you can just use `val = Base.mod(val, mod)`.